Sources/Foundation/JSONSerialization.swift

```diff
@@ -183,22 +183,22 @@ open class JSONSerialization : NSObject {
      */
     open class func jsonObject(with data: Data, options opt: ReadingOptions = []) throws -> Any {
         do {
-            let jsonValue = try data.withUnsafeBytes { (ptr) -> JSONValue in
-                let (encoding, advanceBy) = JSONSerialization.detectEncoding(ptr)
-                
+            let jsonValue = try {
+                let (encoding, advanceBy) = JSONSerialization.detectEncoding(data)
+
                 if encoding == .utf8 {
                     // we got utf8... happy path
-                    var parser = JSONParser(bytes: Array(ptr[advanceBy..<ptr.count]))
+                    var parser = JSONParser(bytes: data[advanceBy...])
                     return try parser.parse()
                 }
-                
-                guard let utf8String = String(bytes: ptr[advanceBy..<ptr.count], encoding: encoding) else {
+
+                guard let utf8String = String(bytes: data[advanceBy...], encoding: encoding) else {
                     throw JSONError.cannotConvertInputDataToUTF8
                 }
-                
-                var parser = JSONParser(bytes: Array(utf8String.utf8))
+
+                var parser = JSONParser(bytes: utf8String.data(using: .utf8)!)
```

Comment:
The force-unwrap makes me wonder about the guarantees here (as they all should). Are we sure this can't fail in any way?